src/parse.d

```diff
@@ -3070,97 +3068,116 @@ public:
 
             if (token.value == TOKlparen)
             {
-                // Class template declaration.
-                // Gather template parameter list
+                // struct/class template declaration.
                 tpl = parseTemplateParameterList();
                 constraint = parseConstraint();
             }
         }
 
-        switch (tok)
+        // Collect base class(es)
+        BaseClasses* baseclasses = null;
+        if (token.value == TOKcolon)
         {
-        case TOKclass:
-        case TOKinterface:
-            {
-                if (!id)
-                    error(loc, "anonymous classes not allowed");
-
-                // Collect base class(es)
-                BaseClasses* baseclasses = null;
-                if (token.value == TOKcolon)
-                {
-                    nextToken();
-                    baseclasses = parseBaseClasses();
-                    if (tpl)
-                    {
-                        Expression tempCons = parseConstraint();
-                        if (tempCons)
-                        {
-                            if (constraint)
-                                error("members expected");
-                            else
-                                constraint = tempCons;
-                        }
-                    }
-                    if (token.value != TOKlcurly)
-                        error("members expected");
-                }
-
-                if (tok == TOKclass)
-                {
-                    bool inObject = md && !md.packages && md.id == Id.object;
-                    a = new ClassDeclaration(loc, id, baseclasses, inObject);
-                }
-                else
-                    a = new InterfaceDeclaration(loc, id, baseclasses);
-                break;
-            }
-        case TOKstruct:
-            if (id)
-                a = new StructDeclaration(loc, id);
-            else
-                anon = 1;
-            break;
-
-        case TOKunion:
-            if (id)
-                a = new UnionDeclaration(loc, id);
-            else
-                anon = 2;
-            break;
+            nextToken();
+            baseclasses = parseBaseClasses();
+        }
 
-        default:
-            assert(0);
+        if (token.value == TOKif)
+        {
+            if (constraint)
+                error("multiple constraints not allowed");
```

Comment:
Can you include the location of the second constraint or is this done automatically?
I can tell from the test, because it looks like it is not compiled with `-vcolumns`.